Library/ViewModels/RewardAddOnSelectionViewModelTests.swift

```diff
@@ -1107,25 +1107,25 @@ final class RewardAddOnSelectionViewModelTests: TestCase {
     self.vm.inputs.configure(with: data)
     self.vm.inputs.viewDidLoad()
 
-    XCTAssertEqual(["Add-Ons Page Viewed"], self.dataLakeTrackingClient.events)
-    XCTAssertEqual(["Add-Ons Page Viewed"], self.segmentTrackingClient.events)
-
     self.scheduler.advance()
     self.vm.inputs.shippingRuleSelected(shippingRule)
     self.scheduler.advance()
 
+    XCTAssertEqual(["Page Viewed"], self.dataLakeTrackingClient.events)
```

Comment:
We shouldn't be moving the line down. We expect the event to be fired after viewDidLoad.

Reply:
Ah and can you include a check for the properties after we check the event?

Reply:
So what fires the "Page Viewed" event is inside an asynchronous task, `XCTAssertEqual(["Page Viewed"], self.dataLakeTrackingClient.events)` is being called immediately after viewDidLoad, so there's a possibility that `XCTAssertEqual(["Page Viewed"], self.dataLakeTrackingClient.events)` will be called before the "Page Viewed" event is fired. 

Moving `XCTAssertEqual(["Page Viewed"], self.dataLakeTrackingClient.events)` to after `self.scheduler.advance()` would ensure that all actions that have been scheduled are executed  as soon as possible.

 I think the fact that it passes previously is a false positive.

Reply:
Not sure I agree with this. If the test was working before then it is fair to assume it was correct. The issue is very likely related to the changes you applied to the view model. Keep in mind many more parameters were introduced to the tracking logic (`RewardAddOnSelectionViewModel line 284`) and that is likely the source of the issue.